azavea/idb-osm-extraction-tool
# Create geocoder

## Overview

This PR implements the geocoder control mostly as depicted in the wireframes.

For the geocoder, we use the Esri world geocoder. I've set up an app under the Azavea organization and I added the relevant domains there.

For coordinate search, we just use the builtin Leaflet L.latLng class to attempt the conversion, doing the work in a Promise so that if it fails we can catch it as normal.

I temporarily switched out the basemap for Positron because the OSM tile server was not responding. I'll drop that commit before merging.

Connects #9

### Demo

![geocoder](https://user-images.githubusercontent.com/4165523/43484887-a26365ba-94dd-11e8-9d49-060bc760a0c9.gif)

![coordinates](https://user-images.githubusercontent.com/4165523/43484891-a579ef4e-94dd-11e8-94ba-691c417af94c.gif)


### Notes

I filtered the geocoder suggestions using a Guyana bbox, then filtered more to remove countries which are not Guyana.

## Testing Instructions

The dev environment has changed since the last time a PR went up against this project, so it's probably best to start by destroying your prior VM and rm -rf the repo, then cloning the repo again and running `./scripts/setup`

Once that's done, serve the app on localhost:4567 and visit the page in the browser in both Chrome and IE 11 and verify

- you can use the geocoder to get results and select from the suggestions to zoom to the aoi
- the coordinate search works and handles errors properly

There's a Netlify site deployed linked below which you can also check.
